tapchannel/aux_invoice_manager_test.go

```diff
@@ -0,0 +1,571 @@
+				{
+					Invoice: &lnrpc.Invoice{
+						RouteHints: testRouteHints(),
+						ValueMsat:  3_000_000,
+					},
+					WireCustomRecords: newWireCustomRecords(
+						t, []*rfqmsg.AssetBalance{
+							rfqmsg.NewAssetBalance(
+								assetID(1), 3,
+							),
+						}, fn.Some(dummyRfqID(31)),
+					),
+				},
+			},
+			responses: []lndclient.InvoiceHtlcModifyResponse{
+				{
+					AmtPaid: 3_000_000,
+				},
+			},
+			buyQuotes: rfq.BuyAcceptMap{
+				dummyRfqID(31).Scid(): {
+					Peer:     testNodeID,
+					AskPrice: 1_000_000,
+				},
+			},
+		},
+		{
+			name: "asset invoice, not enough amt",
+			requests: []lndclient.InvoiceHtlcModifyRequest{
+				{
+					Invoice: &lnrpc.Invoice{
+						RouteHints: testRouteHints(),
+						ValueMsat:  3_000_000,
+					},
+					WireCustomRecords: newWireCustomRecords(
+						t, []*rfqmsg.AssetBalance{
+							rfqmsg.NewAssetBalance(
+								assetID(1), 3,
+							),
+						}, fn.Some(dummyRfqID(31)),
+					),
+					ExitHtlcAmt: 1234,
+				},
+			},
+			responses: []lndclient.InvoiceHtlcModifyResponse{
+				{
+					AmtPaid: 1_500_000,
+				},
+			},
+			buyQuotes: rfq.BuyAcceptMap{
+				dummyRfqID(31).Scid(): {
+					Peer:     testNodeID,
+					AskPrice: 500_000,
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+
+		t.Logf("Running AuxInvoiceManager test case: %v", testCase.name)
+
+		// Instantiate mock rfq manager.
+		mockRfq := &mockRfqManager{
+			peerBuyQuotes:   testCase.buyQuotes,
+			localSellQuotes: testCase.sellQuotes,
+		}
+
+		done := make(chan bool)
+
+		// Instantiate mock htlc modifier.
+		mockModifier := &mockHtlcModifier{
+			requestQue:     testCase.requests,
+			expectedResQue: testCase.responses,
+			done:           done,
+			t:              t,
+		}
+
+		// Create the manager.
+		manager := NewAuxInvoiceManager(
+			&InvoiceManagerConfig{
+				ChainParams:         testChainParams,
+				InvoiceHtlcModifier: mockModifier,
+				RfqManager:          mockRfq,
+			},
+		)
+
+		err := manager.Start()
+		require.NoError(t, err)
+
+		// If the manager is not done processing the htlc modification
+		// requests within the specified timeout, assume this is a
+		// failure.
+		select {
+		case <-done:
+		case <-time.After(testTimeout):
+			t.Fail()
+		}
+	}
+}
+
+// genRandomRfqID generates a random rfqmsg.ID value.
+func genRandomRfqID(t *rapid.T) rfqmsg.ID {
+	var rfqID rfqmsg.ID
+	for i := 0; i < len(rfqID); i++ {
+		rfqID[i] = rapid.Byte().Draw(t, "scid_byte")
+		// rapid.SliceOfN(rapid.Byte(), len(rfqID), len(rfqID))
+	}
+	return rfqID
+}
+
+// genInvoice generates an invoice that may have a random amount, and may have
+// routing hints.
+func genInvoice(t *rapid.T) *lnrpc.Invoice {
+	res := &lnrpc.Invoice{}
+
+	// Do a random draw with a 1/8 chance of introducing an empty invoice.
+	nullInv := rapid.Uint16().Draw(t, "invoice_exists")
+	if nullInv%8 == 0 {
+		return res
+	}
+
+	// Generate a random invoice value.
+	valueMsat := rapid.Int64Range(
```

Comment:
I would recommend making the ranges for these numerical values like invoice value, units, and price much smaller; at least to start. IIUC this allows the test framework to cover more edge cases more quickly (and combinations of these variables that may be edge cases).

IIUC for these tests we really care about the relationship between these three values and not the absolute magnitudes.

Would also be good to have comments explaining why each range is the size it is, and why `0` may or may not be a meaningful value, etc.

Reply:
> IIUC this allows the test framework to cover more edge cases more quickly (and combinations of these variables that may be edge cases).

The rapid framework itself tends to stick more to the ends of specified ranges, in order to expose such edge case behavior.

> IIUC for these tests we really care about the relationship between these three values and not the absolute magnitudes.

Right, I changed the invoice amt / asset units drawn values to have a range which relates to each other. Added some docs on the related section